tests/v2/validation/rbac/rbac_test.go

```diff
@@ -0,0 +1,239 @@
+		fmt.Println("Validation successful!")
+	}
+
+	//Testcase6 Validate if cluster members are able to delete the project they are not owner of
+	fmt.Printf("Testcase6 - Validating if %s can delete a project they are not owner of. \n",role)
+	err = projects.DeleteProject(rb.standardUserClient, createProjectAsAdmin)
+	if role == "cluster-owner"{
+		require.NoError(rb.T(), err)
+		fmt.Println("Validation successful!")
+	}else if role == "cluster-member"{
+	require.Error(rb.T(), err)
+	errStatus := strings.Split(err.Error(), ".")[1]
+	rgx := regexp.MustCompile(`\[(.*?)\]`)
+	errorMsg := rgx.FindStringSubmatch(errStatus)
+	assert.Equal(rb.T(), "403 Forbidden" , errorMsg[1])
+	fmt.Println("Validation successful!")
+	}
+
+	//Testcase7 Validate if cluster members are able to delete the namespace they are not owner of
+	fmt.Printf("Testcase7 - Validating if %s can delete a namespace from a project they are not owner of. \n",role)
+	err = namespaces.DeleteNamespace(rb.standardUserClient,namespaceName, rb.cluster.ID )
+	if role == "cluster-owner"{
+		require.NoError(rb.T(),err)
+		fmt.Println("Validation successful!")
+	}else if role == "cluster-member"{
+		require.Error(rb.T(),err)
+		fmt.Println("Valdiation successful!")
+	}
+
+	//Testcase8 Validate if a standard user return nil list of clusters
+	//Remove added cluster member from the cluster as an admin
+	fmt.Println("Testcase8 - Validating cluster member is removed from the cluster and returns nil clusters")
+	err = users.RemoveClusterMember(rb.client, rb.standardUser)
+	time.Sleep(6 * time.Second)
+	require.NoError(rb.T(), err)
+	fmt.Println("Testcase8 - Validating standard users cannot list any downstream clusters")
+	rb.standardUserClient, err = rb.standardUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+	err = rb.ValidateClustersList(rb.standardUserClient,0, "")
+	require.Error(rb.T(),err)
+	assert.Equal(rb.T(), "Resource type [provisioning.cattle.io.cluster] is not listable", err.Error())
+}
+
+func (rb *rbTestSuite) TestRBAC(){
+	tests := []struct {
+		name string
+		clusterRole string
+	}{
+		{"Cluster Owner", "cluster-owner"},
+		{"Cluster Member", "cluster-member"},
+	}
+	for _, tt := range tests {
+		rb.Run(tt.name, func() {
+			newUser, err := CreateUser(rb.client)
+			require.NoError(rb.T(), err)
+			rb.standardUser = newUser
+			fmt.Println("Created user: ", rb.standardUser.Username)
+			rb.standardUserClient, err = rb.client.AsUser(newUser)
+			require.NoError(rb.T(), err)
+			subSession := rb.session.NewSession()
+			defer subSession.Cleanup()	
+			_, err = rb.standardUserClient.WithSession(subSession)
+			require.NoError(rb.T(), err)
+			//Adding created user to the downstream clusters with the specified roles.
+			fmt.Printf("Adding user as %s to the downstream cluster. \n", tt.name)
+			err = users.AddClusterMember(rb.client, rb.cluster, rb.standardUser, tt.clusterRole)
+			require.NoError(rb.T(), err)
+
+			time.Sleep(6 * time.Second)
```

Comment:
not sure what the sleep is for

Reply:
Similar to the above adding a WatchWait, test errors out as the user is not updated with the permissions. Therefore added the time.Sleep. I'll look more into it probably using an incorrect resource wait.